pkg/sidecarlogresults/sidecarlogresults.go

```diff
@@ -0,0 +1,179 @@
+							now := time.Now().UnixMilli()
+							if _, ok := startTimesPerResult[newResult.Name]; !ok {
+								startTimesPerResult[newResult.Name] = now
+							} else if now-startTimesPerResult[newResult.Name] >= millisecondsTowaitForEmptyResult {
+								// value is still empty after second from the time that the result file was created.
+								resultParsed[newResult.Name] = true
+							}
+						} else {
+							// populated value is not changing anymore after 50 ms.
+							// assume that the contents have been fully written.
+							resultParsed[newResult.Name] = true
+						}
+					}
+				}
+				results[newResult.Name] = newResult.Value
+
+				if resultParsed[newResult.Name] == true {
+					newResult.log()
+					resultsFetched.Insert(v.Name())
+				}
+			}
+		}
+		f.Close()
+		time.Sleep(sleepTime)
+		if reflect.DeepEqual(resultsFetched, expectedResults) {
+			break
+		}
+	}
+	return nil
+}
+
+// GetResultsFromSidecarLogs extracts results from the logs of the results sidecar
+func GetResultsFromSidecarLogs(ctx context.Context, clientset kubernetes.Interface, namespace string, name string, container string, sidecarLogResults []v1beta1.PipelineResourceResult) ([]v1beta1.PipelineResourceResult, error) {
+	podLogOpts := corev1.PodLogOptions{Container: container}
+	req := clientset.CoreV1().Pods(namespace).GetLogs(name, &podLogOpts)
+	sidecarLogs, err := req.Stream(ctx)
+	if err != nil {
+		podInitializingError := fmt.Sprintf(`container "%v" in pod "%v" is waiting to start: PodInitializing`, container, name)
+		if err.Error() == podInitializingError {
+			return sidecarLogResults, nil
+		}
+		return sidecarLogResults, err
+	}
+	defer sidecarLogs.Close()
+	maxResultLimit := config.FromContextOrDefaults(ctx).FeatureFlags.MaxResultSize
+	sidecarLogResults, err = extractResultsFromLogs(sidecarLogs, sidecarLogResults, maxResultLimit)
+	return sidecarLogResults, err
+}
+
+func extractResultsFromLogs(logs io.Reader, sidecarLogResults []v1beta1.PipelineResourceResult, maxResultLimit int) ([]v1beta1.PipelineResourceResult, error) {
+	scanner := bufio.NewScanner(logs)
+	buf := make([]byte, 2)
+	scanner.Buffer(buf, maxResultLimit)
+	for scanner.Scan() {
+		text := scanner.Text()
+		result, err := parseResults(text, maxResultLimit)
+		if err != nil {
+			return sidecarLogResults, err
+		}
+		sidecarLogResults = append(sidecarLogResults, result)
+	}
+
+	if err := scanner.Err(); err != nil {
+		if errors.Is(err, bufio.ErrTooLong) {
+			return sidecarLogResults, ErrorReasonMaxResultSizeExceeded
+		}
+		return sidecarLogResults, err
+	}
+	return sidecarLogResults, nil
+}
+
+func parseResults(resultStr string, maxResultLimit int) (v1beta1.PipelineResourceResult, error) {
+	result := v1beta1.PipelineResourceResult{}
+	if len(resultStr) > maxResultLimit {
+		return result, ErrorReasonMaxResultSizeExceeded
+	}
+
+	var res SidecarLogResult
+	if err := json.Unmarshal([]byte(resultStr), &res); err != nil {
+		return result, fmt.Errorf("Invalid result %v", resultStr)
```

Comment:
```suggestion
		return result, fmt.Errorf("invalid result %w", err)
```

Then annotate with the result name in the caller. The raw result string probably isn't worth logging all the time. If you want, consider moving this into a zap debug log level).

Reply:
Sounds good!